### DISCUSSION TITLE
`definePageMeta` is not importable

### DISCUSSION THREAD

**Discussion by Patrick-Remy** [nuxt/nuxt#16907] [Ideas]
Global available 'auto-imported' types of nuxt3, are in general also explicit importable. This is useful as the LS sometimes doesn't get the types defined inside `.nuxt3/auto-imports.d.ts`. 

Generated types defined `node_modules/nuxt3/dist/app` are importable via 
```ts
import { defineNuxtComponent } from '#app'
// or import { defineNuxtComponent } from 'nuxt3' 
```
`vue`-types are importable via
```ts
import { defineComponent } from 'vue'
```

But `definePageMeta` is declared inside `node_modules/nuxt3/dist/pages/runtime/composables.d.ts` which isn't re-exported.

Therefore I suggest to:
- Like inside `app` directory, create a `pages/types/index.d.ts` that exports at least all types inside `pages/runtime/composables.d.ts`.
- Add `export * from './dist/pages/types/index'` to `nuxt3/types.d.ts`

Then it would be possible to import
```ts
import { definePageMeta } from 'nuxt3' 
```

**Comment by danielroe** [comment]
`definePageMeta` is a compiler macro - it shouldn't be imported as it will have no effect elsewhere in your app.

In general, you can import any auto-importable composable from '#imports', however.

**Reply by szulcus** [reply]
> `definePageMeta` is a compiler macro - it shouldn't be imported as it will have no effect elsewhere in your app.
> 
> In general, you can import any auto-importable composable from '#imports', however.

![image](https://user-images.githubusercontent.com/48328739/189343692-c6f49435-36ef-41f4-ac36-89a486f05c7d.png)
I am using Nuxt 2 + Nuxt Bridge and can't import `definePageMeta`. I also searched the entire directory `node_modules/@nuxt/bridge/dist/runtime` and found no such expression there 🤔.

**Reply by danielroe** [reply]
It is not relevant to Nuxt Bridge; it's a Nuxt 3 feature.

**Reply by szulcus** [reply]
> Nie dotyczy Nuxt Bridge; jest to funkcja Nuxt 3.

I understand. So is there any way to use middleware in `<script setup>`?

**Reply by danielroe** [reply]
No, there is not. You can create a secondary `<script>` block for this purpose.

**Reply by szulcus** [reply]
Ok, thanks for your answer.

**Reply by bete7512** [reply]
well done my problem is gone
